src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -584,33 +584,6 @@ public async Task<HttpResponseMessage> SendAsyncCore(HttpRequestMessage request,
                     ParseStatusLine(await ReadNextResponseHeaderLineAsync().ConfigureAwait(false), response);
                 }
 
-                // If we sent an Expect: 100-continue header, and didn't receive a 100-continue. Handle the final response accordingly.
```

Comment:
Moving this code down below seems like it breaks the 100-continue logic. Right below this code, we await sendRequestContentTask. But since you moved this code, it won't signal the allowExpect100ToContinue task and so we'll always wait for the timer before sending the request body. What am I missing here?

Reply:
yes, I missed that part @geoffkizer and it did not show up in tests. We need to parse headers in order to know if we have NTLM/NEGO so I can move sending request body after parsing headers as well or special case 401 handling and add to possibly revert the _connectionClose=true decision.